### PR TITLE
bump flyteidl version

### PR DESCRIPTION
**Pull request by ckiosidis**
Signed-off-by: Babis Kiosidis <babisk@spotify.com>

# TL;DR

I manually changed the flyteidl to a version (0.21.5) that includes this release https://github.com/flyteorg/flyteidl/releases/tag/v0.21.4 
The run `go mod tidy` before commiting

## Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by ckiosidis** [comment]
thanks @evalsocket
